### PR TITLE
Add 1.16.4 Support

### LINKED ISSUE #54: Implement 1.16.4
As a developer, I would like to use my modifications on the newer versions.

### PR DESCRIPTION
**Pull request by R0bbyYT**
Adds 1.16.4 Support.

Closes #54

### CODE REVIEW

File: mcapi/src/internal/java/net/flintmc/mcapi/internal/render/text/DefaultFontRenderBuilder.java [hunk 1]
```diff
@@ -145,21 +149,48 @@ public FontRenderBuilder disableShadow() {
...
-        this.x,
-        this.y,
-        this.text,
-        this.rgba,
-        this.alignment,
-        this.maxLineLength,
-        this.shadow,
-        this.xFactor,
-        this.yFactor);
+    boolean isNetherUpdate = this.versionHelper.isUnder(16);
```
Comment (5HT2) [inline mcapi/src/internal/java/net/flintmc/mcapi/internal/render/text/DefaultFontRenderBuilder.java]: Personally I think a better name for this variable would be helpful, such as `hasRGBText` (which I assume is what is new, text-wise, unless I'm wrong)

File: mcapi/src/internal/java/net/flintmc/mcapi/internal/render/text/DefaultTooltipRenderBuilder.java [hunk 2]
```diff
@@ -69,12 +73,26 @@ public TooltipRenderBuilder text(String text) {
...
-    this.renderer.renderTooltip(this.x, this.y, this.text);
+    boolean isNetherUpdate = this.versionHelper.isUnder(16);
+
+    if(isNetherUpdate) {
+      Preconditions.checkNotNull(this.matrixStack, "Matrix stack cannot be null!");
+
+      this.renderer.renderTooltip(this.matrixStack, this.x, this.y, this.text);
+    } else {
+      this.renderer.renderTooltip(this.x, this.y, this.text);
+    }
```
Comment (5HT2) [inline mcapi/src/internal/java/net/flintmc/mcapi/internal/render/text/DefaultTooltipRenderBuilder.java]: Formatting is incorrect, make sure inside Editor settings in Intellij you've allowed it to apply style from the `.editorconfig`

File: mcapi/src/internal/java/net/flintmc/mcapi/internal/world/codec/DefaultDatapackCodec.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    this.disabled = disabled;
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public List<String> getEnabled() {
+    return this.enabled;
+  }
+
+  /** {@inheritDoc} */
```
Comment (5HT2) [inline mcapi/src/internal/java/net/flintmc/mcapi/internal/world/codec/DefaultDatapackCodec.java]: Style, as mentioned above

File: mcapi/src/internal/java/net/flintmc/mcapi/internal/world/codec/DefaultDatapackCodecRepository.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+@Implement(DatapackCodecRepository.class)
+public class DefaultDatapackCodecRepository implements DatapackCodecRepository {
+
+  private final Map<String, DatapackCodec> datapackCodecs;
+
+  @Inject
+  private DefaultDatapackCodecRepository(Factory datapackCodecFactory) {
+    this.datapackCodecs = Maps.newConcurrentMap();
+    this.register(
+        "vanilla", datapackCodecFactory.create(ImmutableList.of("vanilla"), ImmutableList.of()));
```
Comment (5HT2) [inline mcapi/src/internal/java/net/flintmc/mcapi/internal/world/codec/DefaultDatapackCodecRepository.java:42]: Iirc the google style has a higher limit on the width of the line, so this should be on one line (just apply the style as mentioned above)

File: mcapi/src/internal/java/net/flintmc/mcapi/internal/world/storage/DefaultWorldOverview.java [hunk 5]
```diff
@@ -19,15 +19,14 @@
...
 import net.flintmc.framework.inject.assisted.Assisted;
 import net.flintmc.framework.inject.assisted.AssistedInject;
 import net.flintmc.framework.inject.implement.Implement;
 import net.flintmc.mcapi.player.type.GameMode;
 import net.flintmc.mcapi.world.storage.WorldOverview;
 
-/**
- * Default implementation of the {@link WorldOverview}.
- */
+/** Default implementation of the {@link WorldOverview}. */
```
Comment (5HT2) [inline mcapi/src/internal/java/net/flintmc/mcapi/internal/world/storage/DefaultWorldOverview.java]: Style :p

File: mcapi/src/main/java/net/flintmc/mcapi/world/codec/DatapackCodec.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package net.flintmc.mcapi.world.codec;
+
+import java.util.List;
+import net.flintmc.framework.inject.assisted.AssistedFactory;
+
+/** Represents a datapack codec. */
```
Comment (5HT2) [inline mcapi/src/main/java/net/flintmc/mcapi/world/codec/DatapackCodec.java]: Style :p

File: mcapi/src/main/java/net/flintmc/mcapi/world/codec/DatapackCodecRepository.java [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package net.flintmc.mcapi.world.codec;
+
+import java.util.Map;
+
+/** Represents a repository to register and unregister {@link DatapackCodec}. */
```
Comment (5HT2) [inline mcapi/src/main/java/net/flintmc/mcapi/world/codec/DatapackCodecRepository.java]: Style

File: mcapi/src/main/java/net/flintmc/mcapi/world/storage/WorldOverview.java [hunk 8]
```diff
@@ -19,13 +19,12 @@
...
 
+import java.io.File;
 import net.flintmc.framework.inject.assisted.Assisted;
 import net.flintmc.framework.inject.assisted.AssistedFactory;
 import net.flintmc.mcapi.player.type.GameMode;
 
-/**
- * Represents an informative interface for the world.
- */
+/** Represents an informative interface for the world. */
```
Comment (5HT2) [inline mcapi/src/main/java/net/flintmc/mcapi/world/storage/WorldOverview.java]: Style

File: mcapi/src/main/java/net/flintmc/mcapi/world/storage/WorldOverview.java [hunk 9]
```diff
@@ -112,30 +118,27 @@
    */
   boolean isOutdatedWorld();
 
-  /**
-   * A factory class for creating {@link WorldOverview}'s.
-   */
+  /** A factory class for creating {@link WorldOverview}'s. */
```
Comment (5HT2) [inline mcapi/src/main/java/net/flintmc/mcapi/world/storage/WorldOverview.java]: Style

File: mcapi/src/main/java/net/flintmc/mcapi/world/storage/WorldOverview.java [hunk 10]
```diff
@@ -112,30 +118,27 @@
...
-     * @param outdatedWorld      {@code true} if the world is outdated, otherwise {@code false}.
+     *     false}.
+     * @param gameMode The game mode of the world.
+     * @param hardcoreMode {@code true} if the hardcore mode is enabled in the world, otherwise
+     *     {@code false}.
+     * @param cheats {@code true} if the cheats are enabled in the world, otherwise {@code false}.
+     * @param askToOpenWorld {@code true} asks a world to open, otherwise {@code false}.
+     * @param markWorldInList {@code true} marks a world in the list, otherwise {@code false}.
+     * @param futureWorld {@code true} if the world in the future, otherwise {@code false}.
+     * @param outdatedWorld {@code true} if the world is outdated, otherwise {@code false}.
```
Comment (5HT2) [inline mcapi/src/main/java/net/flintmc/mcapi/world/storage/WorldOverview.java]: Style

**Review by 5HT2 — CHANGES_REQUESTED** [review]
I'm just going to review this much, not going to bother with the rest. Please make sure your `.editorstyle` is enabled and auto-format the packages which you've edited

File: mcapi/src/internal/java/net/flintmc/mcapi/internal/render/text/DefaultFontRenderBuilder.java (hunk 1 above)
Comment (R0bbyYT) [reply mcapi/src/internal/java/net/flintmc/mcapi/internal/render/text/DefaultFontRenderBuilder.java]: I have renamed it.

**Review by 5HT2 — APPROVED** [review]

**Pull request merged**